src/gmtspatial.c

```diff
@@ -2029,12 +2039,13 @@ EXTERN_MSC int GMT_gmtspatial (void *V_API, int mode, void *args) {
 
 		T = C->table[0];	/* Only one input file so only one table */
 		count = gmt_M_memory (GMT, NULL, D->n_segments, unsigned int);
+		ID = Ctrl->N.ID - 1;	/* May be overridden below by header info, otherwise will be incremented to 0 (or whatever +p set) */
 		for (seg2 = 0; seg2 < T->n_segments; seg2++) {	/* For all polygons */
 			S2 = T->segment[seg2];
 			SH = gmt_get_DS_hidden (S2);
 			if (gmt_polygon_is_hole (GMT, S2)) continue;	/* Holes are handled in gmt_inonout */
 			GMT_Report (API, GMT_MSG_INFORMATION, "Look for points/features inside polygon segment %" PRIu64 " :\n", seg2);
-			if (Ctrl->N.ID == 0) {	/* Look for polygon IDs in the data headers */
+			if (!Ctrl->N.p_set) {	/* Look for polygon IDs in the data headers */
```

Comment:
But if you do this -Z and -L won't be searched in headers unless +p is used.

Reply:
See what I did in #7749

Reply:
Dont get it. If p is not used then Ctrl->N.p_set is false and !false is true and it goes to look in the headers. No?

Reply:
Looks like we are yelling at the same time!